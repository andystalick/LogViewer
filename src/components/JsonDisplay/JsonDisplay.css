.json-display {
  white-space: pre-wrap;
  margin: 0;
  padding: 0.5rem 1rem 0.5rem 1rem;
  background-color: var(--background);
  border-right: 2px solid var(--selected);
  border-bottom: 2px solid var(--selected);
  border-left: 2px solid var(--selected);
  color: var(--code);
}
