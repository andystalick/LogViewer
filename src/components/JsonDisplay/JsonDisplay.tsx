import './JsonDisplay.css';

const JsonDisplay = (props: { str: string }) => {
  const obj = JSON.parse(props.str);
  return (
    <pre className="json-display">
      <code>{JSON.stringify(obj, null, 2)}</code>
    </pre>
  );
};
export default JsonDisplay;
