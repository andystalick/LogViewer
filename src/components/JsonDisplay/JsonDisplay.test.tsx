import { expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import JsonDisplay from './JsonDisplay';

const rawRowFixture =
  '{"_time":1722523685074,"cid":"api","channel":"conf:policies","level":"info","message":"loading policy","context":"cribl","policy":{"args":["groupName","dashboardId"],"template":["GET /m/${groupName}/search/dashboards/${dashboardId}","GET /m/${groupName}/search/dashboards/${dashboardId}/*"],"description":"Can view and use the dashboard","title":"Read Only"}}';

test('snapshot test', async () => {
  render(<JsonDisplay str={rawRowFixture} />);
  expect(screen.getByTestId('json-display')).toMatchSnapshot();
});
