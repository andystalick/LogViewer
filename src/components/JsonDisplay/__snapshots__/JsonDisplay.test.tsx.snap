// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test 1`] = `
<pre
  class="json-display"
  data-testid="json-display"
>
  <code>
    {
  "_time": 1722523685074,
  "cid": "api",
  "channel": "conf:policies",
  "level": "info",
  "message": "loading policy",
  "context": "cribl",
  "policy": {
    "args": [
      "groupName",
      "dashboardId"
    ],
    "template": [
      "GET /m/\${groupName}/search/dashboards/\${dashboardId}",
      "GET /m/\${groupName}/search/dashboards/\${dashboardId}/*"
    ],
    "description": "Can view and use the dashboard",
    "title": "Read Only"
  }
}
  </code>
</pre>
`;
